specification/recoveryservices/Recoveryservices.Management/models.tsp

```diff
@@ -0,0 +1,1685 @@
+
+  /**
+   * Status of the vault upgrade operation.
+   */
+  @visibility(Lifecycle.Read)
+  status?: VaultUpgradeState;
+
+  /**
+   * Message to the user containing information about the upgrade operation.
+   */
+  @visibility(Lifecycle.Read)
+  message?: string;
+
+  /**
+   * The way the vault upgrade was triggered.
+   */
+  @visibility(Lifecycle.Read)
+  triggerType?: TriggerType;
+
+  /**
+   * Resource ID of the upgraded vault.
+   */
+  @visibility(Lifecycle.Read)
+  upgradedResourceId?: string;
+
+  /**
+   * Resource ID of the vault before the upgrade.
+   */
+  @visibility(Lifecycle.Read)
+  previousResourceId?: string;
+}
+
+/**
+ * Information to be stored in Vault properties as an element of privateEndpointConnections List.
+ */
+model PrivateEndpointConnectionVaultProperties {
+  /**
+   * Format of id subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.[Service]/{resource}/{resourceName}/privateEndpointConnections/{connectionName}.
+   */
+  @visibility(Lifecycle.Read)
+  id?: string;
+
+  /**
+   * Private Endpoint Connection Response Properties.
+   */
+  @visibility(Lifecycle.Read)
+  properties?: PrivateEndpointConnection;
+
+  /**
+   * The name of the private Endpoint Connection
+   */
+  @visibility(Lifecycle.Read)
+  name?: string;
+
+  /**
+   * The type, which will be of the format, Microsoft.RecoveryServices/vaults/privateEndpointConnections
+   */
+  @visibility(Lifecycle.Read)
+  type?: string;
+
+  /**
+   * The location of the private Endpoint connection
+   */
+  @visibility(Lifecycle.Read)
+  location?: string;
+}
+
+/**
+ * Private Endpoint Connection Response Properties.
+ */
+model PrivateEndpointConnection {
+  /**
+   * Gets or sets provisioning state of the private endpoint connection.
+   */
+  @visibility(Lifecycle.Read)
+  provisioningState?: ProvisioningState;
+
+  /**
+   * The Private Endpoint network resource that is linked to the Private Endpoint connection.
+   */
+  @visibility(Lifecycle.Read)
+  privateEndpoint?: PrivateEndpoint;
+
+  /**
+   * Gets or sets private link service connection state.
+   */
+  @visibility(Lifecycle.Read)
+  privateLinkServiceConnectionState?: PrivateLinkServiceConnectionState;
+
+  /**
+   * Group Ids for the Private Endpoint
+   */
+  groupIds?: VaultSubResourceType[];
+}
+
+/**
+ * The Private Endpoint network resource that is linked to the Private Endpoint connection.
+ */
+model PrivateEndpoint {
+  /**
+   * Gets or sets id.
+   */
+  @visibility(Lifecycle.Read)
+  id?: string;
+}
+
+/**
+ * Gets or sets private link service connection state.
+ */
+model PrivateLinkServiceConnectionState {
+  /**
+   * Gets or sets the status.
+   */
+  @visibility(Lifecycle.Read)
+  status?: PrivateEndpointConnectionStatus;
+
+  /**
+   * Gets or sets description.
+   */
+  @visibility(Lifecycle.Read)
+  description?: string;
+
+  /**
+   * Gets or sets actions required.
+   */
+  @visibility(Lifecycle.Read)
+  actionsRequired?: string;
+}
```

Comment:
can private endpoint connection definitions be in their own TSP file or better yet, come from common types?

we have common types in swagger.

Reply:
[https://github.com/Azure/azure-rest-api-specs/blob/802ded7f507321d74f4594edf2a78732b83edda6/specification/common-types/resource-management/v5/privatelinks.json#L20](https://github.com/Azure/azure-rest-api-specs/blob/802ded7f507321d74f4594edf2a78732b83edda6/specification/common-types/resource-management/v5/privatelinks.json#L20)

Reply:
I'm sure that moving to best practices is definitely where we want to get to.

Have we spent enough time on the feasibility of incorporating best practices during the conversions yet?

I think it'd be a missed opportunity if during our conversions we don't try to use common-types by default.
Also, even if we don't start recognizing all the cases that *should* be common types, and start filing work items to fix them, where we are blocked.